deployment-server/app/Main.hs

```diff
@@ -0,0 +1,67 @@
+{-# LANGUAGE DeriveAnyClass    #-}
+{-# LANGUAGE DeriveGeneric     #-}
+{-# LANGUAGE NamedFieldPuns    #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards   #-}
+
+module Main where
+
+import           Control.Concurrent       (forkIO)
+import           Control.Concurrent.Chan  (newChan)
+import           Control.Newtype.Generics (Newtype, unpack)
+import           Data.Aeson               (FromJSON, eitherDecodeFileStrict, parseJSON, withObject, (.:))
+import qualified Data.ByteString.Char8    as BS
+import           Data.Map                 as Map
+import           Data.Map                 (Map)
+import           Data.Text                (Text)
+import qualified Data.Text                as Text
+import           Deploy.Server            (app)
+import           Deploy.Worker            (Deployment, SlackChannel, runWorker)
+import           GHC.Generics             (Generic)
+import           Network.Wai.Handler.Warp (run)
+import           Options.Generic          (ParseField, ParseFields, ParseRecord, getRecord)
+import           Servant.GitHub.Webhook   (gitHubKey)
+import           System.IO                (BufferMode (LineBuffering), hSetBuffering, stderr, stdout)
+import           Web.Slack                (mkSlackConfig)
+
+newtype SlackToken = SlackToken Text
+    deriving (Generic, Newtype)
+
+newtype WebhookKey = WebhookKey Text
+    deriving (Generic, Newtype)
+
+data Options = Options
+    { port           :: Int
+    , configDir      :: FilePath
+    , stateFile      :: FilePath
+    , include        :: [String]
+    , keyfile        :: FilePath
+    , slackChannel   :: SlackChannel
+    , deploymentName :: Deployment
+    } deriving (Generic, Show, ParseRecord)
+
+data Secrets = Secrets
+    { githubWebhookKey :: WebhookKey
+    , slackToken       :: SlackToken
+    }
+
+instance FromJSON Secrets where
+    parseJSON =
+        withObject "Secrets" $ \v ->
+            Secrets <$> (WebhookKey <$> v .: "githubWebhookKey") <*> (SlackToken <$> v .: "slackToken")
```

Comment:
Personally, I find nested `<$>` a bridge too far, and would break out a `do` here. Especially as you already have `RecordWildCards` available, so it's just going to end with `pure Secrets {..}`.

Reply:
done.